---
title: 'Özel Durumlar: failwith İşlevi (F#)'
description: "Nasıl 'failwith' işlevi bir F # özel durum oluşturur öğrenin."
author: cartermp
ms.author: phcart
ms.date: 05/16/2016
ms.topic: language-reference
ms.prod: dotnet-fsharp
ms.devlang: fsharp
ms.openlocfilehash: af1e3b7dc96b3b822e2e19b7bac435940d15922f
ms.sourcegitcommit: 03ee570f6f528a7d23a4221dcb26a9498edbdf8c
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 04/28/2018
---
# <a name="exceptions-the-failwith-function"></a><span data-ttu-id="1a6bd-103">Özel Durumlar: failwith İşlevi</span><span class="sxs-lookup"><span data-stu-id="1a6bd-103">Exceptions: The failwith Function</span></span>

<span data-ttu-id="1a6bd-104">`failwith` İşlevi F # özel durum oluşturur.</span><span class="sxs-lookup"><span data-stu-id="1a6bd-104">The `failwith` function generates an F# exception.</span></span>


## <a name="syntax"></a><span data-ttu-id="1a6bd-105">Sözdizimi</span><span class="sxs-lookup"><span data-stu-id="1a6bd-105">Syntax</span></span>

```fsharp
failwith error-message-string
```

## <a name="remarks"></a><span data-ttu-id="1a6bd-106">Açıklamalar</span><span class="sxs-lookup"><span data-stu-id="1a6bd-106">Remarks</span></span>
<span data-ttu-id="1a6bd-107">*Hata ileti dizesi* önceki sözdiziminde sabit değerli bir dize türünde bir değer mi `string`.</span><span class="sxs-lookup"><span data-stu-id="1a6bd-107">The *error-message-string* in the previous syntax is a literal string or a value of type `string`.</span></span> <span data-ttu-id="1a6bd-108">Bu duruma `Message` özel durum özelliği.</span><span class="sxs-lookup"><span data-stu-id="1a6bd-108">It becomes the `Message` property of the exception.</span></span>

<span data-ttu-id="1a6bd-109">Tarafından oluşturulan özel durum `failwith` olan bir `System.Exception` adına sahip bir başvuru özel durumu `Failure` F # kodunda.</span><span class="sxs-lookup"><span data-stu-id="1a6bd-109">The exception that is generated by `failwith` is a `System.Exception` exception, which is a reference that has the name `Failure` in F# code.</span></span> <span data-ttu-id="1a6bd-110">Aşağıdaki kod kullanımını göstermektedir `failwith` bir özel durum için.</span><span class="sxs-lookup"><span data-stu-id="1a6bd-110">The following code illustrates the use of `failwith` to throw an exception.</span></span>

[!code-fsharp[Main](../../../../samples/snippets/fsharp/lang-ref-2/snippet6001.fs)]
    
## <a name="see-also"></a><span data-ttu-id="1a6bd-111">Ayrıca Bkz.</span><span class="sxs-lookup"><span data-stu-id="1a6bd-111">See Also</span></span>
[<span data-ttu-id="1a6bd-112">Özel Durum İşleme</span><span class="sxs-lookup"><span data-stu-id="1a6bd-112">Exception Handling</span></span>](index.md)

[<span data-ttu-id="1a6bd-113">Özel Durum Türleri</span><span class="sxs-lookup"><span data-stu-id="1a6bd-113">Exception Types</span></span>](exception-types.md)

[<span data-ttu-id="1a6bd-114">Özel durumlar: `try...with` ifade</span><span class="sxs-lookup"><span data-stu-id="1a6bd-114">Exceptions: The `try...with` Expression</span></span>](the-try-with-expression.md)

[<span data-ttu-id="1a6bd-115">Özel durumlar: `try...finally` ifade</span><span class="sxs-lookup"><span data-stu-id="1a6bd-115">Exceptions: The `try...finally` Expression</span></span>](the-try-finally-expression.md)

[<span data-ttu-id="1a6bd-116">Özel durumlar: `raise` işlevi</span><span class="sxs-lookup"><span data-stu-id="1a6bd-116">Exceptions: the `raise` Function</span></span>](the-raise-function.md)
